FlashSpace/Features/Gestures/WorkspaceTransitionManager.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  WorkspaceTransitionManager.swift
+//
+//  Created for FlashSpace
+//  Copyright © 2025 Wojciech Kulik. All rights reserved.
+//
+
+import AppKit
+import SwiftUI
+
+final class WorkspaceTransitionManager {
+    static let shared = WorkspaceTransitionManager()
+
+    private var transitionWindow: NSWindow?
+    private var isAnimating = false
+
+    // Configuration
+    var enableTransitionEffects = true
+    var transitionDuration: TimeInterval = 0.3
+
+    func performTransition(direction: TransitionDirection, onComplete: @escaping () -> ()) {
+        guard enableTransitionEffects, !isAnimating else {
+            onComplete()
+            return
+        }
+
+        isAnimating = true
+
+        let contentView = NSHostingView(
+            rootView: WorkspaceTransitionView(direction: direction)
+        )
+
+        // Create a full-screen window that overlays everything
+        guard let mainScreen = NSScreen.main else {
+            isAnimating = false
+            onComplete()
+            return
+        }
+
+        let window = NSWindow(
+            contentRect: mainScreen.frame,
```

Comment:
There is a problem here when using multiple displays. Main screen is the one that has the focus, but you can switch a workspace that is on a different screen, this way the transition appears on the incorrect screen.

However, it complicates things a little, because in the current approach (where you call this function before calling workspace change) you don't know which screen will be updated. 

So I guess you would need to call it from `WorkspaceManager.activateworkspace`, but actually maybe it's even better because probably just this one place should be enough to cover all cases, it would also cover CLI calls.